docs/USAGE.md

```diff
@@ -22,64 +56,341 @@ and edit:
 
-If you use Bitcoin Core's multiwallet feature, you can edit the value of rpc_wallet_file to your chosen wallet file.
+If you use Bitcoin Core's multiwallet feature, you can edit the value of `rpc_wallet_file` to your chosen wallet file.
 
-Then retry the same `generate` command; it should now not error (see below).
+Then retry the same `generate` command; it should now not error (see [below](#generate)).
 If you still get rpc connection errors, make sure you can connect to your Core node using the command line first.
 
-### First step: make a Joinmarket wallet
+<a name="wallet-tool" />
+
+## Using the `wallet-tool.py` script
+
+This section leads you through the basics of creating, funding, recovering and displaying information about the Joinmarket wallet using the `wallet-tool.py` script. A reminder that, at this point, you should be on the command line and in the `scripts/` subdirectory of the Joinmarket directory.
+
+The Joinmarket wallet is hierarchical and deterministic, it can be entirely recovered from a single seed. At any point, you can use `python wallet-tool.py --help` on the command line for all options. We'll now go through all the main functions:
+
+<a name="generate" />
+
+### Creating a Wallet
+Run wallet-tool.py with the `generate` method. Write down the 12 word seed on paper.
+
+        (jmvenv)$ python wallet-tool.py generate
+        Write down this wallet recovery mnemonic on paper:
+
+        matter aim multiply december stove march wolf nuclear yard boost worth supreme
+
+        Enter wallet encryption passphrase: 
+        Reenter wallet encryption passphrase: 
+        Input wallet file name (default: wallet.jmdat): 
+        saved to wallet.jmdat
+        $
+
+So far you have just created the wallet. But to do the next step, you'll need to see the addresses.
+Run `python wallet-tool.py wallet.jmdat` (or replace the wallet name with whatever you chose). You'll get this message:
+
+   **restart Bitcoin Core with -rescan or use `bitcoin-cli rescanblockchain` if you're recovering an existing wallet from backup seed**
+   **Otherwise just restart this joinmarket application.**
+
+Notice that here you **don't** need to rescan, because you are not "recovering an existing wallet" but creating a new one. Because it is new, none of the addresses are used yet. You only need to rescan (or use `rescanblockchain` as mentioned, which by the way can be a more efficient way of rescanning) if the addresses *have* been used ...
+
+<a name="funding" />
+
+### Funding Wallet and Displaying Balance
+
+... so, ignore this, and just run `python wallet-tool.py wallet.jmdat` again. You should get a large output like this:
+
+```
+2019-08-23 19:38:36,030 [DEBUG]  got used indices: {0: [0, 0], 1: [0, 0], 2: [0, 0], 3: [0, 0], 4: [0, 0]}
+2019-08-23 19:38:36,059 [DEBUG]  Wallet successfully synced
+2019-08-23 19:38:36,060 [DEBUG]  rpc: listunspent [0]
+2019-08-23 19:38:37,015 [DEBUG]  bitcoind sync_unspent took 0.9556303024291992sec
```

Comment:
Maybe remove `[DEBUG]` lines from sample outputs? Default `console_log_level` is `INFO`.

Reply:
Fixed, thanks